### DISCUSSION TITLE
PROBLEM IN INSTALLING TEXWORKS

### DISCUSSION THREAD

**Discussion by gavrielsegre** [TeXworks/texworks#1051] [Q&A] [answered]
Good morning

when I install Texworks and I run the program I get the following error message:

"Impossible to find the point of entering D3D12SerializeVersionedRootSignature of the procedure in the library of dynamic linking C:/Program Files (x86)\TexWorks\TexWorks.exe".

Could someone very kindly help me?

**Comment by stloeffler** [comment]
Hi,

On 23.04.24 13:22, gavrielsegre wrote:
> when I install Texworks and I run the program I get the following 
> error message:
>
> "Impossible to find the point of entering 
> D3D12SerializeVersionedRootSignature of the procedure in the library 
> of dynamic linking C:/Program Files (x86)\TexWorks\TexWorks.exe".
>

Please describe the exact version/file of TeXworks you obtained, how you 
obtained it, and which operating system you use.

Background: such errors may be caused by incompatible versions, e.g. 
installing a version requiring Windows 10 or later on - say - a Windows 
7 machine.

Kind regards,
Stefan

**Comment by gavrielsegre** [comment, marked as answer]
Dear Stefan
I got TexWorks version 0.6.9 from  https://www.tug.org/texworks/
downloading it on a Windows 10 operating system.
Thank you very much for your help!


Il giorno sab 27 apr 2024 alle ore 15:43 Stefan Löffler <
***@***.***> ha scritto:

> Hi,
>
> On 23.04.24 13:22, gavrielsegre wrote:
> > when I install Texworks and I run the program I get the following
> > error message:
> >
> > "Impossible to find the point of entering
> > D3D12SerializeVersionedRootSignature of the procedure in the library
> > of dynamic linking C:/Program Files (x86)\TexWorks\TexWorks.exe".
> >
>
> Please describe the exact version/file of TeXworks you obtained, how you
> obtained it, and which operating system you use.
>
> Background: such errors may be caused by incompatible versions, e.g.
> installing a version requiring Windows 10 or later on - say - a Windows
> 7 machine.
>
> Kind regards,
> Stefan
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/TeXworks/texworks/discussions/1051#discussioncomment-9245385>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH35GF4CE5QAP5CNS7GAF5DY7OTRPAVCNFSM6AAAAABGUV2WOSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TENBVGM4DK>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
   Gavriel Segre

**Reply by stloeffler** [reply]
OK, this helps already. The error message about  `D3D12SerializeVersionedRootSignature` indicates it's a problem with finding a DirectX 12 function, which is used by the Qt framework used for managing and displaying all the windows. From searching the internet, DirectX functionality seems to be changing quite rapidly. Therefore, it would be interesting to know your exact Windows version/build number. This you can find, e.g., by running the `winver` program. According to the [D3D12 docs](https://learn.microsoft.com/de-de/windows/win32/api/d3d12/nf-d3d12-d3d12serializeversionedrootsignature), the function in question was added in the 14393 build, which [corresponds to](https://learn.microsoft.com/en-us/windows/release-health/release-information) the 1607 version.
If you have an older Windows version, you should run Windows update. If it's newer, maybe the D3D functionality has changed, which should then be reported to the Qt devs.

In any case, a workaround could be to download the [legacy Win 7 version](https://github.com/TeXworks/texworks/releases/tag/release-0.6.9). This is functionally equivalent, should not rely on DirectX 12, but of course is not optimal as it uses an outdated version of some libraries.

**Comment by gavrielsegre** [comment]
Thanks for the tip!

Il Dom 5 Mag 2024, 08:05 Stefan Löffler ***@***.***> ha
scritto:
